packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -198,113 +197,59 @@ def _get_field(mcs, cls: Type, field_name: str) -> Any:
         return getattr(cls, field_name)
 
 
+@dataclass(frozen=True)
 class BaseTxPayload(ABC, metaclass=_MetaPayload):
     """This class represents a base class for transaction payload classes."""
 
-    transaction_type: Any
-
-    def __init__(
-        self,
-        sender: str,
-        id_: Optional[str] = None,
-        round_count: int = ROUND_COUNT_DEFAULT,
-        **kwargs: Any,
-    ) -> None:
-        """
-        Initialize a transaction payload.
-
-        :param sender: the sender (Ethereum) address
-        :param id_: the id of the transaction
-        :param round_count: the count of the round in which the payload was sent
-        :param kwargs: the keyword arguments
-        """
-        self.id_ = uuid.uuid4().hex if id_ is None else id_
-        self._round_count = round_count
-        self._sender = sender
-        # kwargs exists for typing only
-        enforce(len(kwargs) == 0, f"kwargs passed to {type(self)} must be empty")
+    sender: str
+    transaction_type: Any = field(init=False)
+    round_count: int = field(default=ROUND_COUNT_DEFAULT, init=False)
+    id_: str = field(default_factory=lambda: uuid.uuid4().hex, init=False)
 
     @property
-    def sender(self) -> str:
-        """Get the sender."""
-        return self._sender
+    def data(self) -> Dict[str, Any]:
+        """Data"""
+        excluded = ["sender", "round_count", "id_", "transaction_type"]
+        return {k: v for k, v in asdict(self).items() if k not in excluded}
 
+    # TODO: refactor - these methods are not strictly needed
     @property
-    def round_count(self) -> int:
-        """Get the round count."""
-        return self._round_count
-
-    @round_count.setter
-    def round_count(self, round_count: int) -> None:
-        """Set the round count."""
-        self._round_count = round_count
-
-    def encode(self) -> bytes:
-        """Encode the payload."""
-        return DictProtobufStructSerializer.encode(self.json)
-
-    @classmethod
-    def decode(cls, obj: bytes) -> "BaseTxPayload":
-        """Decode the payload."""
-        return cls.from_json(DictProtobufStructSerializer.decode(obj))
+    def json(self) -> Dict[str, Any]:
+        """Json"""
+        data = asdict(self)
+        data["transaction_type"] = str(data["transaction_type"])
+        return data
```

Comment:
serialization can be dealt with here (class currently not used)
https://github.com/valory-xyz/open-autonomy/blob/22dcb96cbb033ad0c6e7109f15c70c5141250239/packages/valory/skills/abstract_round_abci/base.py#L305-L344